config/tests/src/test/java/org/jboss/windup/config/RuleProviderOverrideTest.java

```diff
@@ -0,0 +1,144 @@
+
+@RunWith(Arquillian.class)
+public class RuleProviderOverrideTest
+{
+
+    @Inject
+    private GraphContextFactory factory;
+    @Inject
+    private RuleLoader loader;
+
+    @Deployment
+    @AddonDependencies({
+                @AddonDependency(name = "org.jboss.windup.config:windup-config"),
+                @AddonDependency(name = "org.jboss.windup.graph:windup-graph"),
+                @AddonDependency(name = "org.jboss.forge.furnace.container:cdi")
+    })
+    public static AddonArchive getDeployment()
+    {
+        return ShrinkWrap.create(AddonArchive.class).addBeansXML();
+    }
+
+    @Test
+    public void testOverride() throws IOException
+    {
+        try (GraphContext context = factory.create())
+        {
+            Predicate<RuleProvider> predicate = new Predicate<RuleProvider>()
+            {
+                @Override
+                public boolean accept(RuleProvider provider)
+                {
+                    return true;
+                }
+            };
+
+            Configuration configuration = loader.loadConfiguration(context, predicate).getConfiguration();
+            int count = 0;
+            for (Rule rule : configuration.getRules())
+            {
+                count++;
+                Assert.assertTrue("Override", rule.toString().contains("RuleOverride"));
+            }
+            Assert.assertEquals(1, count);
+        }
+    }
+
+    @Singleton
+    public static class TestOriginalProvider extends AbstractRuleProvider
+    {
+        public TestOriginalProvider()
+        {
+            super(MetadataBuilder.forProvider(TestOriginalProvider.class, "TestRuleProvider"));
+        }
+
+        @Override
+        public Configuration getConfiguration(GraphContext context)
+        {
+            return ConfigurationBuilder.begin()
+                    .addRule(new Rule()
+                    {
+                        @Override
+                        public void perform(Rewrite event, EvaluationContext context)
+                        {
+                        }
+
+                        @Override
+                        public boolean evaluate(Rewrite event, EvaluationContext context)
+                        {
+                            return true;
+                        }
+
+                        @Override
+                        public String getId()
+                        {
+                            return TestOriginalProvider.class.getSimpleName();
+                        }
+
+                        @Override
+                        public String toString() {
+                            return "OriginalRule";
+                        }
+                    });
+        }
+
+        @Singleton
+        public static class TestOverrideProvider extends AbstractRuleProvider {
+            public TestOverrideProvider() {
+                super(MetadataBuilder.forProvider(TestOverrideProvider.class, "TestRuleProvider").setOverrideProvider(true));
+            }
```

Comment:
That is a new test. I'm not sure why that it would?

On Fri, Apr 15, 2016 at 12:15 PM, Ondrej Zizka <notifications@github.com>
wrote:

> In
> config/tests/src/test/java/org/jboss/windup/config/RuleProviderOverrideTest.java
> <https://github.com/windup/windup/pull/869#discussion_r59899856>:
>
> > +                        {
> > +                            return TestOriginalProvider.class.getSimpleName();
> > +                        }
> > +
> > +                        @Override
> > +                        public String toString() {
> > +                            return "OriginalRule";
> > +                        }
> > +                    });
> > +        }
> > +
> > +        @Singleton
> > +        public static class TestOverrideProvider extends AbstractRuleProvider {
> > +            public TestOverrideProvider() {
> > +                super(MetadataBuilder.forProvider(TestOverrideProvider.class, "TestRuleProvider").setOverrideProvider(true));
> > +            }
>
> This will collide with my PR (just a note)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/windup/windup/pull/869/files/007dd959043b2c5c3385c914013b3c6f12558063#r59899856>
>